x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx

```diff
@@ -443,6 +445,18 @@ export function AlertDetailsAppSection({
             </EuiFlexItem>
           </EuiFlexGroup>
         </EuiFlexItem>
+        <EuiFlexItem grow={false}>
```

Comment:
I noticed that `transactionType` is still retrieve from the rule which looks like a mistake to me. We should get `transaction.type` from the alert. If it's `undefined` (shouldn't be but possibly could happen for legacy alerts) we should not show the preview.

So I suggest changing:
```ts
  const transactionTypes = useServiceTransactionTypesFetcher({
    serviceName,
    start,
    end,
  });
  const transactionType = getTransactionType({
    transactionType: alert.fields[TRANSACTION_TYPE],
    transactionTypes,
    agentName,
  });
```

to this:

```ts
const transactionType = alert.fields[TRANSACTION_TYPE];
```

Reply:
Thank you @sqren!

Reply:
Thank you @sqren!